spec/services/pipeline_report_service_spec.rb

```diff
@@ -137,6 +161,11 @@
 
       expect(JSON.parse(@report)["counts"]["2"]["570"]).to include_json(genus_result)
     end
+
+    it "should show correct values in CSV" do
+      csv_report = PipelineReportService.call(@pipeline_run, @background.id, csv: true)
+      expect(csv_report).to eq("tax_id,tax_level,genus_tax_id,name,common_name,category,agg_score,max_z_score,nt_z_score,nt_rpm,nt_count,nt_contigs,nt_contig_r,nt_percent_identity,nt_alignment_length,nt_e_value,nt_bg_mean,nt_bg_stdev,nr_z_score,nr_rpm,nr_count,nr_contigs,nr_contig_r,nr_percent_identity,nr_alignment_length,nr_e_value,nr_bg_mean,nr_bg_stdev,species_tax_ids\n570,2,570,Klebsiella,,bacteria,2428411764.7058825,99,99,193404.63458110517,217,6,594,99.7014,149.424,89.5822,18.3311,64.2056,99,77540.10695187165,87,6,594,97.9598,46.4253,16.9874,35.0207,238.639,[573]\n573,1,570,Klebsiella pneumoniae,,bacteria,2428411764.7058825,99,99,186274.50980392157,209,2,198,99.6995,149.402,89.5641,9.35068,26.4471,99,61497.326203208555,69,2,198,97.8565,46.3623,16.9101,29.9171,236.332,\n")
```

Comment:
not sure how better to show long string

Reply:
A slight improvement is to define the a constant above.
Side note: A better (also more code) test would be to split into multiple check. Parse the csv and expect correct CSV, expect containing each header (loop), expect contain correct values, etc.